### PR TITLE
DEVDOCS-4413: [update] Subscription Foundation, add Vercel support

### PR DESCRIPTION
**Pull request by bc-traciporter**
# [DEVDOCS-4413](https://bigcommercecloud.atlassian.net/browse/DEVDOCS-4413)

## What changed?
Updates based on the ticket requirements

### CODE REVIEW

File: docs/api-docs/partner/subscriptions/foundation-guide.md [hunk 1]
```diff
@@ -206,19 +207,25 @@ Open the `.env` file you just created and update the following environment varia
...
+1. If you're using Postgres, which is the default data store for Subscription Foundation, skip to the next step. Otherwise, follow the instructions in [Replacing Postgres](#replacing-postgres).
 
-2. Run the pre-configured Prisma migration script to create the database tables and initial client as defined in `/prisma/migrations/*` with the following command: 
+2. Obtain the database URL in [Supabase](https://supabase.com/) by going to **Settings** > **Database** > **Connection string** and clicking **URI**.  Use this string to update the `DATABASE_URL` environment variable in the `.env` file. The string resembles the following example:
+
+```shell title="Example Postgres Cloud connection string"
+postgresql://postgres:[YOUR-PASSWORD]@db.uqchmyniufaqkijttavq.supabase.co:5432/postgres
+```
+
+3. Run the pre-configured Prisma migration script to create the database tables and initial client as defined in `/prisma/migrations/*` with the following command:
```
Comment (becomevocal) [inline docs/api-docs/partner/subscriptions/foundation-guide.md:222]: This also happens automatically now when you `npm run build`.

File: docs/api-docs/partner/subscriptions/foundation-guide.md [hunk 2]
```diff
@@ -195,6 +195,7 @@ Open the `.env` file you just created and update the following environment varia
 | Environment variable | Description | Reference location |
```
Comment (becomevocal) [inline docs/api-docs/partner/subscriptions/foundation-guide.md:195]: Can you also add a section for deploying to Vercel? When doing that, you only need to set the DATABASE_URL to start, with the remaining variable set as placeholders (i.e. using '1234' for each)

**Review by becomevocal — CHANGES_REQUESTED** [review]

File: docs/api-docs/partner/subscriptions/foundation-guide.md (hunk 2 above)
Comment (bc-traciporter) [reply docs/api-docs/partner/subscriptions/foundation-guide.md:195]: @becomevocal thank you for this information. I updated the article. Can you please review?

File: docs/api-docs/partner/subscriptions/foundation-guide.md [hunk 3]
```diff
@@ -274,6 +293,24 @@ To add new subscription rules and edit existing ones, complete the following ste
 
 If you plan to use the API to add products to the subscription sales channel, see [product channel assignments](/api-docs/multi-storefront/api-guide#products) for more information.
 
+## Deploying with Vercel
+
+To deploy with Vercel, click on the **Deploy** button in the [README.md](https://github.com/bigcommerce/subscription-foundation/blob/main/README.md) file, login to GitHub if you have not done so, and complete the following steps:
```
Comment (stanzikratelbc) [inline docs/api-docs/partner/subscriptions/foundation-guide.md:298]: Replace the first sentence with a brief intro: "The BigCommerce Subscription Foundation framework enables you to deploy your application with Vercel directly from the GitHub repo. 

To deploy your app with Vercel:

1. Log into GitHub, and then navigate to the [BigCommerce Subscription Foundation Readme]((https://github.com/bigcommerce/subscription-foundation/blob/main/README.md) file.
2. Scroll down the page to the **Deploy with Vercel** section and click **Deploy**.
3. In the **Get started** section, select a Git provider, and then authorize Vercel to connect to your provider.
4. In the **Create Git Repository** section, select a Git scope, enter a name for the repository, and then click **Create**.
5. (Optional) If you are deploying to Vercel from a GitHub organization, you need to create a team by following the steps in the **Create a team** section.
6. In the **Configure Project** section, in the **Database_URL** field enter the connection string for the Supabase database you created in the previous step. See..... 
7. Enter '1234' for the remaining environment variables, and then click **Deploy**. The deployment process will take a few minutes.
8. After you have successfully deployed Vercel, it is recommended that you update your environment variables. For more information, see....

File: docs/api-docs/partner/subscriptions/foundation-guide.md [hunk 4]
```diff
@@ -284,8 +321,16 @@ This error can occur when the `/prisma/schema.prisma` file contains the incorrec
 
 If you don't enable the proper OAuth scopes, the `/api/auth` request might fail. Go to the [Dev Portal](https://devtools.bigcommerce.com) to verify the proper OAuth scopes are enabled in your app's profile. To learn more, read [Managing Apps in the Developer Portal](/api-docs/apps/guide/developer-portal#edit-technical-details). Make sure your scopes equal or exceed those listed in the [Creating an app profile](#create-an-app-profile) section.
 
+### Issue: Unable to migrate your data to another provider. 
+
+In order to manually switch the database provider, you must do the following:
```
Comment (stanzikratelbc) [inline docs/api-docs/partner/subscriptions/foundation-guide.md:326]: Change intro to: "To manually switch the database provider, complete the following steps:"

**Review by stanzikratelbc — COMMENTED** [review]
Added some feedback. I recommend adding some additional screenshots as well for the Vercel steps

File: docs/api-docs/partner/subscriptions/foundation-guide.md [hunk 5]
```diff
@@ -163,7 +163,7 @@ After you successfully configure test mode, configure your BigCommerce sandbox s
 
 To configure the store to make subscription charges, complete the following steps:
 
-1. In the BigCommerce store control panel, navigate to the [Stripe settings section](https://login.bigcommerce.com/deep-links/settings/payment/stripev3) and make sure that **Test Mode** is set to **Yes**.
+1. In the BigCommerce store control panel, navigate to **Store Setup** > **Payments** > [Stripe settings](https://login.bigcommerce.com/deep-links/settings/payment/stripev3) and make sure that **Test Mode** is set to **Yes**.
```
[the remark this answers is not in the document]
Comment (bc-traciporter) [reply docs/api-docs/partner/subscriptions/foundation-guide.md:166]: @stanzikratelbc I think the path to the Stripe settings tab might be different if you have MLI enabled. Do you have MLI enabled on your store?

File: docs/api-docs/partner/subscriptions/foundation-guide.md [hunk 6]
```diff
@@ -228,19 +239,27 @@ npm run dev
...
+
+2. In `/prisma/.env`, update the value of the `DATABASE_URL` variable to match your new database connection string. Use this string to update the `DATABASE_URL` environment variable in the `.env` file.
 
 3. Run the prisma migration script with the following command:
 
 ```shell title="Run Prisma migration"
 npx prisma migrate dev
 ```
+<!-- theme: warning -->
+> If you miss the preceding step, you will not successfully switch the database provider. For a list of Prisma migrate limitations, see [Prisma Migrate limitations and known issues](https://www.prisma.io/docs/concepts/components/prisma-migrate/prisma-migrate-limitations-issues#you-cannot-automatically-switch-database-providers).
```
[the remark this answers is not in the document]
Comment (bc-traciporter) [reply docs/api-docs/partner/subscriptions/foundation-guide.md:262]: @stanzikratelbc the change makes this sentence passive. That's why I was reluctant to write it this way. I am wondering if there is a better way to write this. Thinking...

File: docs/api-docs/partner/subscriptions/foundation-guide.md [hunk 7]
```diff
@@ -228,19 +239,27 @@ npm run dev
...
 After the app server and ngrok are running, you can install the draft app on your sandbox store. For more information about installing and troubleshooting apps in development, read our App Tutorial section on [Installing and launching an app](/api-docs/apps/tutorials/sample-app-nextjs/step-2-connect#install-and-launch-the-app).
 
-## Replacing SQLite
+## Replacing Postgres
 
-To use an alternate SQL database, complete the following steps:
+To use an alternate SQL database (e.g., SQLite), complete the following steps:
 
-1. Update the `/prisma/schema.prisma` file with a `provider` other than `SQLite`. For a list of options, read [Prisma's reference docs](https://prisma.io/docs/reference/tools-and-interfaces/prisma-schema/data-sources/).
+1. Update the `/prisma/schema.prisma` file with a `provider` other than `Postgresql`. For a list of options, read [Prisma's reference docs](https://prisma.io/docs/reference/tools-and-interfaces/prisma-schema/data-sources/).
```
[the remark this answers is not in the document]
Comment (markcmurphy) [reply docs/api-docs/partner/subscriptions/foundation-guide.md:246]: Looks like this link redirects to https://www.prisma.io/docs/concepts/components/prisma-schema/data-sources

File: docs/api-docs/partner/subscriptions/foundation-guide.md [hunk 8]
```diff
@@ -7,7 +7,7 @@ Subscription Foundation uses the [Channels toolkit](/api-docs/channels/guide/ove
 ## Software requirements 
 * [Node.js](https://nodejs.org/en/) 14.17.0
 * The [npm](https://www.npmjs.com/) package manager
-* A [supported SQL database engine](https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-schema/data-sources/), either SQLite or another of your choice
+* A [supported SQL database engine](https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-schema/data-sources/), either Postgres or another database server of your choice
```
[the remark this answers is not in the document]
Comment (markcmurphy) [reply docs/api-docs/partner/subscriptions/foundation-guide.md:10]: Updated link is https://www.prisma.io/docs/concepts/components/prisma-schema/data-sources

File: docs/api-docs/partner/subscriptions/foundation-guide.md (hunk 3 above)
Comment (bc-traciporter) [reply docs/api-docs/partner/subscriptions/foundation-guide.md]: @stanzikratelbc Working on getting a screenshot for item #5

File: docs/api-docs/partner/subscriptions/foundation-guide.md [hunk 9]
```diff
@@ -274,6 +293,27 @@ To add new subscription rules and edit existing ones, complete the following ste
...
+2. Scroll down the page to the **Deploy with Vercel** section and click **Deploy**.
+3. In the **Get started** section, select a Git provider, and then authorize Vercel to connect to your provider.
+4. In the **Create Git Repository**, select a Git scope, enter a name for the repository, and then click **Create**. 
+5. (Optional) If you are deploying to Vercel from a GitHub organization, you need to create a team by following the steps in the **Create a team** section.
+6. In the **Configure Project** section, enter the connection string for the Supabase database you created in the previous step in the **DATABASE_URL** field. See [Run migration and start the server](#run-migration-and-start-the-server) for information on creating the database and obtaining the `DATABASE_URL`.
+7. Enter '1234' for the remaining environment variables and then click **Deploy**. The deployment process will take a few minutes.
+8. After you have successfully deployed Vercel, it is recommended that you update your environment variables. For more information, see [Declare environment variables](#declare-environment-variables).
```
Comment (markcmurphy) [inline docs/api-docs/partner/subscriptions/foundation-guide.md:309]:
```suggestion
8. After you have successfully deployed to Vercel, it is recommended that you update your environment variables. For more information, see [Declare environment variables](#declare-environment-variables). 
```
More than recommended, if you want the app to integrate properly then you must go back and enter in the rest of them now.

File: docs/api-docs/partner/subscriptions/foundation-guide.md [hunk 10]
```diff
@@ -274,6 +293,27 @@ To add new subscription rules and edit existing ones, complete the following ste
...
+5. (Optional) If you are deploying to Vercel from a GitHub organization, you need to create a team by following the steps in the **Create a team** section.
+6. In the **Configure Project** section, enter the connection string for the Supabase database you created in the previous step in the **DATABASE_URL** field. See [Run migration and start the server](#run-migration-and-start-the-server) for information on creating the database and obtaining the `DATABASE_URL`.
+7. Enter '1234' for the remaining environment variables and then click **Deploy**. The deployment process will take a few minutes.
+8. After you have successfully deployed Vercel, it is recommended that you update your environment variables. For more information, see [Declare environment variables](#declare-environment-variables). 
+
+<!-- theme: info -->
+> #### Updating `NEXT_PUBLIC_APP_URL`
+> Vercel generates the `NEXT_PUBLIC_APP_URL` after the first deployment. You can update the APP_URL with this value.
+
+9. Re-deploy Vercel for the changes to take place. Click on the **Deployments** tab and select **Redeploy** from the three vertical dots on the right.
```
Comment (markcmurphy) [inline docs/api-docs/partner/subscriptions/foundation-guide.md:315]:
```suggestion
9. Redeploy your application to Vercel for the environment variable changes to take place. Click on the **Deployments** tab and select **Redeploy** from the three vertical dots on the right.
```

File: docs/api-docs/partner/subscriptions/foundation-guide.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (slsriehl) [reply docs/api-docs/partner/subscriptions/foundation-guide.md:246]: @bc-traciporter I found the new real link for POS Foundations. Check it out.

**Pull request merged**